src/fundus/scraping/common_crawl/pipeline.py

```diff
@@ -0,0 +1,274 @@
+        One can specify a date range from <start> to <end> to fetch only articles crawled in this range.
+        The default range is 2016/8 -> datetime.now()
+        This corresponds to the crawl date of the CC-News crawler, not the publishing date.
+        To filter on publishing dates use the <only_complete> parameter and refer to the docs about
+        filtering articles.
+
+        Args:
+            start: (datetime): Earliest possible crawl date for retrieved articles. Defaults to 2016/8/1
+            end: (datetime): Latest possible crawl date for retrieved articles. Defaults to datetime.now()
+            max_articles (Optional[int]): Number of articles to crawl. If there are fewer articles
+                than max_articles the Iterator will stop before max_articles. If None, all retrievable
+                articles are returned. Defaults to None.
+            error_handling (Literal["suppress", "catch", "raise"]): Define how to handle errors
+                encountered during extraction. If set to "suppress", all errors will be skipped, either
+                with None values for respective attributes in the extraction or by skipping entire articles.
+                If set to "catch", errors will be caught as attribute values or, if an entire article fails,
+                through Article.exception. If set to "raise" all errors encountered during extraction will
+                be raised. Defaults to "suppress".
+            only_complete (Union[bool, ExtractionFilter]): Set a callable satisfying the ExtractionFilter
+                protocol as an extraction filter or use a boolean. If False, all articles will be yielded,
+                if True, only those with all attributes extracted. Defaults to ExtractionFilter letting
+                through all articles with at least title, body, and publishing_date set.
+            url_filter (Optional[URLFilter]): A callable object satisfying the URLFilter protocol to skip
+                URLs before download. This filter applies on both requested and responded URL. Defaults to None.
+            only_unique (bool): If set to True, articles yielded will be unique on the responded URL.
+                Always returns the first encountered article. Defaults to True.
+
+        Returns:
+            Iterator[Article]: An iterator yielding objects of type Article.
+        """
+
+        if max_articles == 0:
+            return
+        elif max_articles is None:
+            max_articles = -1
+
+        def build_extraction_filter() -> Optional[ExtractionFilter]:
+            if isinstance(only_complete, bool):
+                return (
+                    None
+                    if only_complete is False
+                    else lambda extracted: not all(
+                        bool(v) if not isinstance(v, Exception) else False for _, v in extracted.items()
+                    )
+                )
+            else:
+                return only_complete
```

Comment:
Correct, but as i wrote earlier this is in the somewhat wired state of being refactored after we removed asyncio.

Reply:
Yes, it would not be easy to refactor the affected code directly with other code. But in this circumstance, it would be a simple function. But I'm fine with including it with later refactorings.